src/test/regress/citus_tests/test/test_other_databases.py

```diff
@@ -0,0 +1,140 @@
+def test_main_commited_outer_not_yet(cluster):
+    c = cluster.coordinator
+
+    # create a non-main database
+    c.sql("CREATE DATABASE db1")
+
+    # we will use cur1 to simulate non-main database user and
+    # cur2 to manually do the steps we would do in the main database
+    with c.cur(dbname="db1") as cur1:
+        with c.cur() as cur2:
+
+            # let's start a transaction and find its transaction id
+            cur1.execute("BEGIN")
+            cur1.execute("SELECT txid_current()")
+            txid = cur1.fetchall()
+
+            # using the transaction id of the cur1 simulate the main database commands manually
+            cur2.execute("BEGIN")
+            cur2.execute(
+                "SELECT citus_internal.start_management_transaction('"
+                + str(txid[0][0])
+                + "')"
```

Comment:
This is both easier to read (imho) and better practice, because it avoids sql injection (not that that really matters here, but good practice in general not to interpolate sql strings manually when not needed).

```suggestion
            cur2.execute(
                "SELECT citus_internal.start_management_transaction(%s)",
                str(txid[0][0]))
```